packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/android/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,447 @@
+            is IdxClientResult.Error -> {
+                // handle errors
+            }
+            is IdxClientResult.Success -> {
+                client = clientResult.result
+                // calls the IDX API resume to receive the first IDX response.
+                when (val resumeResult = clientResult.result.resume()) {
+                    is IdxClientResult.Error -> {
+                        // handle errors
+                    }
+                    is IdxClientResult.Success -> {
+                        // handle the response `resumeResult.result`
+                        handleResponse(resumeResult.result)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+The `IdxClientConfigurationProvider` class used in the call to `IdxClient.start()` contains the configuration information for your Okta org, and is usually initialized from an `okta.properties` file. The call to `clientResult.result.resume()` requests the first response of the sign-sequence.
+
+### 3: Authenticate the user credentials
+
+The SDK uses an `IdxResponse` object to represent the server response for the current sign-on step. It contains a collection of remediations (`IdxRemediation`) that specify the type of step, any general messages, and a nested collection of forms (`IdxRemediation.Form`) containing fields (`IdxRemediation.Form.Field`) that represent the various elements required for the UI, such as an text field with a label or a QR code. Completing a step usually requires setting the value of one or more fields. The remediation is then sent back to the server with the updated values by calling the `proceed()` function. The server processes the values and then sends the next step and the the cycle continues until the user is authenticated.
+
+The kind of sign-on step is represented by the `type` property of the remediation. In the first step the server is requesting the username and password, an `IDENTIFY` remediation type.
+
+This code shows the `handleResponse` function that handles the an SDK response indicating a next step and includes all the types of remediations that are part of a Google Authenticator flow:
+
+```kotlin
+private suspend fun handleResponse(response: IdxResponse) {
+    // If a response is successful, immediately exchange it for a token and set it on LoggedInUserView
+    if (response.isLoginSuccessful) {
+        when (val exchangeCodesResult = client?.exchangeInteractionCodeForTokens(response.remediations[ISSUE]!!)) {
+            is IdxClientResult.Error -> {
+                // handle error
+            }
+            is IdxClientResult.Success -> {
+               // handle success and get tokens from `exchangeCodesResult.result`
+            }
+        }
+        return
+    }
+
+    // Check for messages, such as entering an incorrect code or auth error and abort if there is message.
+    if (response.messages.size != 0) {
```

Comment:
There's lots of nice kotlin things that can be used for stuff like this.

`response.messages.isNotEmpty()`. This applies to other areas of this code block as well.

Reply:
again habbits from other languages :rofl: i'll update thanks